rs/test_utilities/execution_environment/src/lib.rs

```diff
@@ -275,6 +275,21 @@ impl ExecutionTest {
             .unwrap()
     }
 
+    pub fn canister_is_wasm64(&self, canister_id: CanisterId) -> bool {
+        // state(), then canister_state(), then execution_state() need to be Some().
+        self.state.as_ref().map_or(false, |state| {
+            state
+                .canister_state(&canister_id)
+                .as_ref()
+                .map_or(false, |canister| {
+                    canister
+                        .execution_state
+                        .as_ref()
+                        .map_or(false, |execution_state| execution_state.is_wasm64)
+                })
+        })
```

Comment:
Nit: Maybe this is a case where chaining too much obscures readability. What if we break it down? Dumb code is at least easy to read.

Reply:
thanks, I tried to make it more clear now.